modular_zubbers/code/modules/mob/check_pulse.dm

```diff
@@ -0,0 +1,27 @@
+/mob/verb/check_pulse(atom/A as mob in view())
+	set name = "Check Pulse"
+	set category = "Object"
+
+	DEFAULT_QUEUE_OR_CALL_VERB(VERB_CALLBACK(src, PROC_REF(_check_pulse), A))
+
+
+/mob/proc/_check_pulse(mob/atomic)
+	if(client && !(atomic in view(client.view, src)) && !client in orange(1, atomic))
+		return FALSE
+	balloon_alert(client.mob, "checking pulse...")
+	if(do_after(client.mob, 4 SECONDS, atomic))
+		if(isliving(client.mob))
+			to_chat(atomic, span_userdanger("You feel [client.mob] grasp you, feeling you closely!"))
+		if(atomic.stat == DEAD || HAS_TRAIT(atomic, TRAIT_NOBREATH) || HAS_TRAIT(atomic, TRAIT_NOBLOOD))
+			balloon_alert(client.mob, "no pulse!")
+			return TRUE
+		if(atomic.stat == HARD_CRIT)
+			balloon_alert(client.mob, "weak pulse!")
+			return TRUE
+		if(atomic.stat == SOFT_CRIT)
+			balloon_alert(client.mob, "erratic pulse!")
+			return TRUE
+
```

Comment:
Would probably be faster to first check all the different traits/edge cases (And there are some you missed, like the MOB_ROBOTIC biotype), then put the stat checks into a switch case

Reply:
Robotics can get organic limbs or some combination thereof, so I don't want to assume too much